dm/worker/worker.go

```diff
@@ -662,3 +703,194 @@ func (w *Worker) MigrateRelay(ctx context.Context, binlogName string, binlogPos
 	}
 	return nil
 }
+
+// copyConfigFromWorker copies config items from dm-worker to sub task
+func (w *Worker) copyConfigFromWorker(cfg *config.SubTaskConfig) {
+	cfg.From = w.cfg.From
+
+	cfg.Flavor = w.cfg.Flavor
+	cfg.ServerID = w.cfg.ServerID
+	cfg.RelayDir = w.cfg.RelayDir
+	cfg.EnableGTID = w.cfg.EnableGTID
+
+	// we can remove this from SubTaskConfig later, because syncer will always read from relay
+	cfg.AutoFixGTID = w.cfg.AutoFixGTID
+
+	// log config items, mydumper unit use it
+	cfg.LogLevel = w.cfg.LogLevel
+	cfg.LogFile = w.cfg.LogFile
+}
+
+// findSubTask finds sub task by name
+func (w *Worker) findSubTask(name string) *SubTask {
+	w.RLock()
+	defer w.RUnlock()
+
+	return w.subTasks[name]
+}
+
+func (w *Worker) restoreSubTask() error {
+	tasks := w.meta.LoadTaskMeta()
+
+	w.Lock()
+	defer w.Unlock()
+	w.subTasks = make(map[string]*SubTask)
+	for name, task := range tasks {
+		taskCfg := new(config.SubTaskConfig)
+		if err := taskCfg.Decode(string(task.Task)); err != nil {
+			return errors.Annotate(err, "decode subtask config error in restoreSubTask")
+		}
+
+		cfgDecrypted, err := taskCfg.DecryptPassword()
+		if err != nil {
+			return errors.Trace(err)
+		}
+
+		log.Infof("[worker] start sub task with config: %v", cfgDecrypted)
```

Comment:
the decrypted password will be logged. the same in L828.

Reply:
`json encode` wouldn't contain password in String() function?

Reply:
oh, that's right.